besu/src/main/java/org/hyperledger/besu/services/TraceServiceImpl.java

```diff
@@ -88,61 +91,108 @@ public void traceBlock(final Hash hash, final BlockAwareOperationTracer tracer)
     block.ifPresent(value -> trace(value, tracer));
   }
 
-  private void trace(final Block block, final BlockAwareOperationTracer tracer) {
-    LOG.debug("Tracing block {}", block.toLogString());
-    final List<TransactionProcessingResult> results = new ArrayList<>();
+  /**
+   * Traces range of blocks
+   *
+   * @param fromBlockNumber the beginning of the range (inclusive)
+   * @param toBlockNumber the end of the range (inclusive)
+   * @param beforeTracing Function which performs an operation on a MutableWorldState before tracing
+   * @param afterTracing Function which performs an operation on a MutableWorldState after tracing
+   * @param tracer an instance of OperationTracer
+   */
+  @Override
+  public void trace(
+      final long fromBlockNumber,
+      final long toBlockNumber,
+      final Consumer<WorldState> beforeTracing,
+      final Consumer<WorldState> afterTracing,
+      final BlockAwareOperationTracer tracer) {
+    checkArgument(tracer != null);
```

Comment:
it might be wise to implement some rational limit on the number or size of the blocks we are tracing.  If only to prevent an unintentional DoS of besu with super long block ranges

Reply:
I think this method is private and only used internally with plugin API. This one is too much in the deep layers. I think if we want to add a limitation it need to be done and managed before calling this one (RPC part etc). 

Also close to impossible to define a good limit. it will depends on the block size and the network configuration .